File: practice/educational_DP/1st.cpp
#include <bits/stdc++.h>
using ll = long long int;
using namespace std;

int main()
{
    int N;
    cin >> N;
    vector<int> h(N), dp(N, 1000000000);
    for (int i = 0; i < N; i++)
    {
        cin >> h[i];
    }
    dp[0] = 0;
    for (int i = 1; i < N; i++)
    {
        dp[i] = min(dp[i], dp[i - 1] + abs(h[i] - h[i - 1]));
        if (i > 1)
        {
            dp[i] = min(dp[i], dp[i - 2] + abs(h[i] - h[i - 2]));
        }
    }
    cout << dp[N - 1] << endl;

    return 0;
}

File: practice/educational_DP/1st.py
N = int(input())
H = list(map(int, input().split()))
dp = [10**10 for i in range(0, N)]
dp[0] = 0
dp[1] = min(dp[1], dp[0]+abs(H[0]-H[1]))
for i in range(2, N):
    dp[i] = min(dp[i], dp[i-1]+abs(H[i]-H[i-1]))
    dp[i] = min(dp[i], dp[i-2]+abs(H[i]-H[i-2]))
print(dp[N-1])
